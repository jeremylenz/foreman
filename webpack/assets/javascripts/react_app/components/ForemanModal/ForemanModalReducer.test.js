import Immutable from 'seamless-immutable';
import { testReducerSnapshotWithFixtures } from '../../common/testHelpers';
import reducer from './ForemanModalReducer';
import {
  SET_MODAL_OPEN,
  SET_MODAL_CLOSED,
  ADD_MODAL,
} from './ForemanModalConstants';

const modalsState = Immutable({
  myModal: { open: true },
  yourModal: { open: false },
});

const fixtures = {
  'initial state': modalsState,
  'should handle SET_MODAL_OPEN action': {
    state: modalsState,
    action: { type: SET_MODAL_OPEN, payload: { id: 'myModal' } },
  },
  'should handle SET_MODAL_CLOSED action': {
    state: modalsState,
    action: { type: SET_MODAL_CLOSED, payload: { id: 'myModal' } },
  },
  'should add a modal with ADD_MODAL': {
    action: { type: ADD_MODAL, payload: { id: 'modal3' } },
  },
  'should add an already-open modal with ADD_MODAL': {
    action: { type: ADD_MODAL, payload: { id: 'modal4', open: true } },
  },
};

describe('ForemanModal reducer', () => {
  describe('ADD_MODAL', () => {
    it('does not create duplicate modals', () => {
      expect(
        reducer(modalsState, {
          type: ADD_MODAL,
          payload: { id: 'yourModal' },
        })
      ).toEqual(modalsState);
    });
    it('is idempotent and does not alter state of existing modal', () => {
      expect(
        reducer(modalsState, {
          type: ADD_MODAL,
          payload: { id: 'myModal', open: false }, // try to add an already existing modal with a different open state
        })
      ).toEqual(modalsState);
    });
  });

  describe('SET_MODAL_OPEN', () => {
    it('throws an error if you try to open a nonexistent modal', () => {
      expect(() => {
        reducer(modalsState, {
          type: SET_MODAL_OPEN,
          payload: { id: 'doesNotExist' },
        });
      }).toThrow('does not exist');
    });
  });

  describe('SET_MODAL_CLOSED', () => {
    it('throws an error if you try to close a nonexistent modal', () => {
      expect(() => {
        reducer(modalsState, {
          type: SET_MODAL_CLOSED,
          payload: { id: 'doesNotExist' },
        });
      }).toThrow('does not exist');
    });
  });

  describe('snapshots', () =>
    testReducerSnapshotWithFixtures(reducer, fixtures));
});
