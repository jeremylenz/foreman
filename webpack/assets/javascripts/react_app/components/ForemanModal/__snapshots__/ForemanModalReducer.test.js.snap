// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanModal reducer snapshots initial state 1`] = `Object {}`;

exports[`ForemanModal reducer snapshots should add a modal with ADD_MODAL 1`] = `
Object {
  "modal3": Object {
    "open": false,
  },
}
`;

exports[`ForemanModal reducer snapshots should add an already-open modal with ADD_MODAL 1`] = `
Object {
  "modal4": Object {
    "open": true,
  },
}
`;

exports[`ForemanModal reducer snapshots should handle SET_MODAL_CLOSED action 1`] = `
Object {
  "myModal": Object {
    "open": false,
  },
  "yourModal": Object {
    "open": false,
  },
}
`;

exports[`ForemanModal reducer snapshots should handle SET_MODAL_OPEN action 1`] = `
Object {
  "myModal": Object {
    "open": true,
  },
  "yourModal": Object {
    "open": false,
  },
}
`;
